src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetLegendGraphicTest.java

```diff
@@ -78,6 +95,42 @@ public void testEnv() throws Exception {
     }
     
     /**
+     * Tests an custom legend graphic
+     */
+    @Test
+    public void testCustomLegend() throws Exception {
+        String base = "wms?service=WMS&version=1.1.1&request=GetLegendGraphic" +
+                        "&layer=sf:states&style=custom" +
+                        "&format=image/png&width=22&height=22";
+        
+        BufferedImage image = getAsImage(base, "image/png");        
+        Resource resource = getResourceLoader().get("styles/legend.png");
+        BufferedImage expected = ImageIO.read( resource.file() );
+        
+        assertEquals( getPixelColor(expected,10,2).getRGB(), getPixelColor(image,10,2).getRGB() );
+        
+        // test rescale
+        base = "wms?service=WMS&version=1.1.1&request=GetLegendGraphic" +
+                "&layer=sf:states&style=custom" +
+                "&format=image/png&width=16&height=16";
+
+        image = getAsImage(base, "image/png");        
+        
+        String user_home = System.getProperty("user.home");
+        File home = new File(user_home);
+        ImageIO.write(expected, "png",  new File(home,"expected.png"));
```

Comment:
Thanks Ben, sorry we missed that in such a long lived patch.
On Wed, Nov 4, 2015 at 4:16 AM Ben Caradoc-Davies <notifications@github.com>
wrote:

> In
> src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetLegendGraphicTest.java
> <https://github.com/geoserver/geoserver/pull/551#discussion_r43839044>:
>
> > +        BufferedImage image = getAsImage(base, "image/png");
> > +        Resource resource = getResourceLoader().get("styles/legend.png");
> > +        BufferedImage expected = ImageIO.read( resource.file() );
> > +
> > +        assertEquals( getPixelColor(expected,10,2).getRGB(), getPixelColor(image,10,2).getRGB() );
> > +
> > +        // test rescale
> > +        base = "wms?service=WMS&version=1.1.1&request=GetLegendGraphic" +
> > +                "&layer=sf:states&style=custom" +
> > +                "&format=image/png&width=16&height=16";
> > +
> > +        image = getAsImage(base, "image/png");
> > +
> > +        String user_home = System.getProperty("user.home");
> > +        File home = new File(user_home);
> > +        ImageIO.write(expected, "png",  new File(home,"expected.png"));
>
> @jodygarnett <https://github.com/jodygarnett> @tbarsballe
> <https://github.com/tbarsballe> please do not write test images to user
> home directory. Fixed in 4380977
> <https://github.com/geoserver/geoserver/commit/438097734641e2d99a2346d186179b0bd0338e08>
> on master. These images were not used in the test and I assume that they
> were for test development and inadvertently included in this PR. At the
> least, ensure that test images are confined to module target so they can be
> cleaned. See gt-render for examples of how to write test fixtures that
> enable interactive development of tests and by default write no files.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoserver/geoserver/pull/551/files#r43839044>.
>
-- 
--
Jody Garnett